### ISSUE TITLE
Catastrophic error when using intelliSense

### ISSUE THREAD

**Issue by tausbn** [github/vscode-codeql#856]
**Describe the bug**
(This is perhaps needlessly vague, but I really have no idea what might be causing this.)

Sometimes, when I type stuff in the CodeQL extension, it starts spitting out catastrophic errors on each keypress. The errors disappear if I remove the offending code-in-progress.

<details><summary>The catastrophic error in question</summary>
<p>

```
May 05, 2021 2:25:18 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint fallbackResponseError
SEVERE: Internal error: com.semmle.util.exception.CatastrophicError: Tried to create dependency to null binding
java.util.concurrent.CompletionException: com.semmle.util.exception.CatastrophicError: Tried to create dependency to null binding
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: com.semmle.util.exception.CatastrophicError: Tried to create dependency to null binding
	at com.semmle.frontend.analysis.QLReferenceGraphBuilder.addDependency(QLReferenceGraphBuilder.java:231)
	at com.semmle.frontend.analysis.QLReferenceGraphBuilder.addDependency(QLReferenceGraphBuilder.java:209)
	at com.semmle.frontend.analysis.QLReferenceGraphBuilder.visit(QLReferenceGraphBuilder.java:663)
	at com.semmle.frontend.ast.HigherOrderTerm.acceptImpl(HigherOrderTerm.java:49)
	at com.semmle.frontend.ast.ASTNode.accept(ASTNode.java:16)
	at com.semmle.frontend.ast.ClasslessPredicate.acceptImpl(ClasslessPredicate.java:31)
	at com.semmle.frontend.ast.ASTNode.accept(ASTNode.java:16)
	at com.semmle.frontend.ast.Module.acceptImpl(Module.java:85)
	at com.semmle.frontend.ast.ASTNode.accept(ASTNode.java:16)
	at com.semmle.frontend.ast.Module.acceptImpl(Module.java:73)
	at com.semmle.frontend.ast.ASTNode.accept(ASTNode.java:16)
	at com.semmle.frontend.analysis.QLReferenceGraphBuilder.getDependencyGraph(QLReferenceGraphBuilder.java:156)
	at com.semmle.frontend.analysis.QLStratificationReporter.checkStratification(QLStratificationReporter.java:31)
	at com.semmle.frontend.Binder.createProgramBinding(Binder.java:179)
	at com.semmle.frontend.CachingBinder.bindingFor(CachingBinder.java:155)
	at com.semmle.frontend.CachingBinder.getBindingFor(CachingBinder.java:103)
	at com.semmle.frontend.Binder.getBindingFor(Binder.java:61)
	at com.semmle.frontend.project.QLProject.lambda$documentSymbol$9(QLProject.java:453)
	... 4 more
```

</p>
</details> 

**Version**
CodeQL Extension v. 1.4.7
CodeQL CLI v. 2.5.4

(Aside: the bug report instructions here say 'To copy version information for the CodeQL extension, click "CodeQL CLI vX.X.X" in the status bar at the bottom of the screen.', however clicking doesn't copy the version information but rather attempts to open https://codeql.github.com/docs/.)

**To reproduce**
I have found the following to be a hopefully consistent way of reproducing the error. I was able to reproduce it in a fresh Codespaces instance by using the following instructions:

1. Go to the file `python/ql/src/semmle/python/ApiGraphs.qll`.
2. Go to line 451 (just after the definition of `trackUseNode/1`) and add the following code
```ql
    private DataFlow::LocalSourceNode trackDefNode(DataFlow::LocalSourceNode nd,
    DataFlow::TypeBackTracker t) {
      t.start() and
      any() and
      result = nd.getALocalSource()
      or
      none()
    }
```
3. Now, remove the `none()` and start typing `exists(DataFlow::TypeBackTracker t)`. As soon as you get to the `::` part, it should start spitting out errors.

**Expected behavior**
An absence of catastrophic errors.

**Additional context**
N/A

**Comment by alexet** [comment]
Smaller reproduction:
```
predicate x() = fastTC(y/2)()
```

**Comment by adityasharad** [comment]
Fix is shipped in CodeQL CLI 2.5.5.
